src/functions-and-arrays.js

```diff
@@ -1,31 +1,63 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if(num1 > num2){
+    return num1;
+  }
+  else {
+    return num2;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(words) {
+  let result = words[0];
+  
+  if(words.length === 0){
+    return null;
+  } 
+  else {
+    for(let i = 0; i < words.length; i++){ 
+        if (words[i].length > result.length){
+          result = words[i];
+        }  
+    }
+  
+  return result;
+}
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
-
-function sumNumbers() {}
-
-
+function sumNumbers(numbers) {
+  let sum = 0;
+  for(let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
+const  mixedArr  =  [ 6 ,  12 ,  'miami' ,  1 ,  true ,  'barca' ,  '200' ,  'lisboa' ,  8 ,  10 ];
+function sum(mixedArr) {
+  let suma = 0;
+  for(let i = 0; i < mixedArr; i++){
```

Comment:
`mixedArr.length` en un for para recorrer un array debemos de usar el length del array